src/librustc_mir/dataflow/impls/storage_liveness.rs

```diff
@@ -43,9 +43,14 @@ impl<'a, 'tcx> BitDenotation<'tcx> for MaybeStorageLive<'a, 'tcx> {
     }
 
     fn terminator_effect(&self,
-                         _sets: &mut BlockSets<'_, Local>,
-                         _loc: Location) {
-        // Terminators have no effect
+                         sets: &mut BlockSets<'_, Local>,
+                         loc: Location) {
+        match &self.mir[loc.block].terminator().kind {
+            TerminatorKind::Drop { location, .. } => if let Some(l) = location.local() {
```

Comment:
Oh, is this the change in the semantics I saw discussed on-thread?
This code should at least have a big comment explaining why exactly we have to consider `Drop`s *that* powerful.
Also, I thought `DropAndReplace` expanded to a `Drop` and an assignment, wouldn't this make that assignment UB?

Reply:
Oh this was merged in #60840.

Reply:
cc @RalfJung

Reply:
I don't know, I've asked you specifically about what you are suggesting the semantics for `DropAndReplace` are and got no reply yet. ;)

I also do not know where these arise, Miri never sees them. Let's discuss in https://github.com/rust-lang/rust/issues/61015 ?

Reply:
Yeah this is incorrect for `DropAndReplace`, I'll have a follow up PR soon and am adding a test to this PR.